src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -447,14 +603,10 @@ public override bool MoveNext()
                         state = 2;
                         goto case 2;
                     case 2:
-                        while (_enumerator.MoveNext())
+                        if (_enumerator.MoveNext())
```

Comment:
I was expecting to see here code that used ```List<T>``` indexer rather than getting an enumerator and using MoveNext/Current.  Is that not good for some reason?